lib/Mojolicious/Guides/Testing.pod

```diff
@@ -0,0 +1,754 @@
+
+The L<Mojo::UserAgent> object in L<Test::Mojo> will know where the application
+is running and make requests to it. Once the tests have completed, the
+L<Mojolicious> application will be torn down.
+
+  my $t = Test::Mojo->new('Frogs');  ## runs on localhost:32114
+
+This object initializes a L<Mojo::UserAgent> object, loads the Mojolicious
+application C<Frogs>, binds and listens on a free TCP port (32114 in this
+example), and starts the application event loop. When the L<Test::Mojo> object
+(C<$t>) goes out of scope, the application is stopped.
+
+Relative URLs in the test object method assertions (C<get_ok>, C<post_ok>,
+etc.) will be sent to the Mojolicious application started by L<Test::Mojo>:
+
+  $t->get_ok('/frogs');  ## http://localhost:32114/frogs
+
+L<Test::Mojo> has a lot of handy shortcuts built into it to make testing
+L<Mojolicious> or L<Mojolicious::Lite> applications enjoyable.
+
+=head3 An Example
+
+Let's spin up a Mojolicious application using C<mojo generate app MyApp>. The
+C<mojo> utility will create a working application and a C<t> directory with a
+working test file:
+
+  $ mojo generate app MyApp
+  [mkdir] /my_app/script
+  [write] /my_app/script/my_app
+  [chmod] /my_app/script/my_app 744
+  ...
+  [mkdir] /my_app/t
+  [write] /my_app/t/basic.t
+  ...
+
+Let's run the tests (we'll create the C<log> directory to quiet the
+application output):
+
+  $ cd my_app
+  $ mkdir log
+  $ prove -lv t
+  t/basic.t ..
+  ok 1 - GET /
+  ok 2 - 200 OK
+  ok 3 - content is similar
+  1..3
+  ok
+  All tests successful.
+  Files=1, Tests=3,  0 wallclock secs ( 0.03 usr  0.01 sys +  0.33 cusr  0.07 csys =  0.44 CPU)
+  Result: PASS
+
+The boilerplate test file looks like this:
+
+  use Mojo::Base -strict;
+
+  use Test::More;
+  use Test::Mojo;
+
+  my $t = Test::Mojo->new('MyApp');
+  $t->get_ok('/')->status_is(200)->content_like(qr/Mojolicious/i);
+
+  done_testing();
+
+Here we can see our application class name C<MyApp> is passed to the
+L<Test::Mojo> constructor. Under the hood, L<Test::Mojo> creates a new
+L<Mojo::Server> instance, loads C<MyApp> (which we just created), and runs the
+application. We write our tests with relative URLs because L<Test::Mojo> takes
+care of getting the request to the running test application (since its port
+may change between runs).
+
+=head3 Testing with Configuration Data
+
+We can alter the behavior of our application using environment variables (such
+as C<MOJO_MODE>) and through configuration values. One nice feature of
+L<Test::Mojo> is its ability to pass configuration values directly from its
+constructor.
+
+Let's modify our application to enable a new feature when the C<enable_test>
+configuration value is set:
+
+  # Load configuration from hash returned by "my_app.conf"
+  my $config = $self->plugin('Config');
+
+  # Normal route to controller
+  $r->get('/')->to('example#welcome');
+
+  # NEW: this route only exists when 'enable_test' is set in the configuration
```

Comment:
This is basically a "feature flag", though I didn't call it that. Feature flags allow you to soft-target an audience without doing a full release. Once it's proved out, you refactor the conditional and turn it into a release. Perhaps if I changed the word 'test' and 'testing' in the URL and flag to something like 'feature' would help clarify that? I didn't want to conflate release management with testing, only illustrate how nice it is that Test::Mojo lets you pass in config on the fly.

Perhaps I'll add another example with a database since both you and Stefan asked for that.